tools/c7n_gcp/tests/test_spanner.py

```diff
@@ -53,6 +55,174 @@ def test_spanner_instance_get(self):
+        self.assertEqual(instances[1]['displayName'], non_patching_instance_name)
+        self.assertEqual(instances[0]['nodeCount'], 1)
+        self.assertEqual(instances[1]['nodeCount'], 1)
+
+    def test_spanner_instance_set_iam_policy_add(self):
+        """
+        The case combines all 3 possible ways for the members to form within a single role:
+        - the members specified in a policy already exist in a resource;
+        - there are existing members in addition to the ones specified in the policy;
+        - a new role is added.
+        """
+        project_id = 'custodian-test-project-0'
+        resource_name = 'spanner-instance-0'
+        resource_full_name = 'projects/%s/instances/%s' % (project_id, resource_name)
+        session_factory = self.replay_flight_data(
+            'spanner-instance-set-iam-policy-add', project_id=project_id)
+        policy = self.load_policy(
+            {'name': 'spanner-instance-set-iam-policy-add',
+             'resource': 'gcp.spanner-instance',
+             'actions': [{'type': 'set-iam-policy',
+                          'add-bindings':
+                              [{'members': ['user:yauhen_shaliou@comelfo.com'],
+                                'role': 'roles/owner'},
+                               {'members': ['user:alex.karpitski@gmail.com'],
+                                'role': 'roles/viewer'},
+                               {'members': ['user:mediapills@gmail.com'],
+                                'role': 'roles/editor'}
+                               ]}]},
+            session_factory=session_factory)
+
+        client = policy.resource_manager.get_client()
+        actual_bindings = client.execute_query('getIamPolicy', {'resource': resource_full_name})
+        self.assertEqual(actual_bindings['bindings'],
+                         [{'members': ['user:yauhen_shaliou@comelfo.com'],
+                           'role': 'roles/owner'},
+                          {'members': ['user:dkhanas@gmail.com'],
+                           'role': 'roles/viewer'}])
+
+        resources = policy.run()
+        self.assertEqual(len(resources), 1)
+        self.assertEqual(resources[0]['name'], resource_full_name)
+
+        if self.recording:
+            time.sleep(1)
+
+        actual_bindings = client.execute_query('getIamPolicy', {'resource': resource_full_name})
+        self.assertEqual(actual_bindings['bindings'],
+                         [{'members': ['user:mediapills@gmail.com'],
+                           'role': 'roles/editor'},
+                          {'members': ['user:yauhen_shaliou@comelfo.com'],
+                           'role': 'roles/owner'},
+                          {'members': ['user:alex.karpitski@gmail.com',
+                                       'user:dkhanas@gmail.com'],
+                           'role': 'roles/viewer'}])
+
+    def test_spanner_instance_set_iam_policy_remove(self):
```

Comment:
note for future, we don't need to re-test every use of this set-iam-policy implementation its pretty normalized across resources in its utilization.

Reply:
Thank you, we will follow the recommendation